### PR TITLE
[valloxmv] Added channels re. filter change date

### PR DESCRIPTION
**Pull request by bjoernbrings**
Added channels in valloxmv binding regarding filter change date (triggered by user request in [forum](https://community.openhab.org/t/new-binding-vallox-mv-ventilation-unit-series-beta/36735/82))

### CODE REVIEW

**Comment by bjoernbrings** [comment]
@wborn I have a (locally) working build to migrate to bnd (following your PRs). Shall I include that migration in this PR or do a seperate one when this one is done?

File: addons/binding/org.openhab.binding.valloxmv/src/main/java/org/openhab/binding/valloxmv/internal/ValloxMVWebSocket.java [hunk 1]
```diff
@@ -257,41 +261,50 @@ public void onBinary(InputStream in) {
...
-                int bdSuppFanBalanceBase = getNumber(bytes, 376);
+                int iRemainingTimeForFilter = getNumber(bytes, 236);
+                int iFilterChangedDateDay = getNumber(bytes, 496);
+                int iFilterChangedDateMonth = getNumber(bytes, 498);
+                int iFilterChangedDateYear = getNumber(bytes, 500);
 
-                int bdHomeSpeedSetting = getNumber(bytes, 418);
-                int bdAwaySpeedSetting = getNumber(bytes, 406);
-                int bdBoostSpeedSetting = getNumber(bytes, 430);
+                Calendar cal = Calendar.getInstance();
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.valloxmv/src/main/java/org/openhab/binding/valloxmv/internal/ValloxMVWebSocket.java]: `cal` is a very weak name for a variable. Let's use something more identifying here like `cFilterChangedDate`. Wdyt?

File: addons/binding/org.openhab.binding.valloxmv/ESH-INF/thing/thing-types.xml [hunk 2]
```diff
@@ -172,6 +174,21 @@
...
+	<channel-type id="filterchangeddate" advanced="true">
+		<item-type>DateTime</item-type>
+		<label>Last filter change</label>
+		<description>Date filter was changed last time</description>
+		<category>Date</category>
+		<state readOnly="true" />
+	</channel-type>
+
+	<channel-type id="remainingfilterdays" advanced="true">
+		<item-type>Number</item-type>
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.valloxmv/ESH-INF/thing/thing-types.xml]: You can use a default item type with support for `QuantityType` here: `Number:Time`.

**Comment by wborn** [comment]
Thanks for helping out with the migration @bjoernbrings! I also think a separate PR would be better. This one will probably be merged soon. :-)

File: addons/binding/org.openhab.binding.valloxmv/ESH-INF/thing/thing-types.xml [hunk 3]
```diff
@@ -172,6 +174,21 @@
 		<state pattern="%d h" readOnly="true" />
 	</channel-type>
 
+	<channel-type id="filterchangeddate" advanced="true">
+		<item-type>DateTime</item-type>
+		<label>Last filter change</label>
+		<description>Date filter was changed last time</description>
+		<category>Date</category>
```
Comment (wborn) [inline addons/binding/org.openhab.binding.valloxmv/ESH-INF/thing/thing-types.xml]: The "Date" category doesn't exist so UIs would not know how to handle it. There is a "Time" category. See [categories](https://www.eclipse.org/smarthome/documentation/concepts/categories.html) on how they are used and what categories are available.

File: addons/binding/org.openhab.binding.valloxmv/ESH-INF/i18n/valloxmv_de.properties [hunk 4]
```diff
@@ -49,6 +49,10 @@ channel-type.valloxmv.uptimehours.label = Zeit in Betrieb (Stunden)
 channel-type.valloxmv.uptimehours.description = Betriebszeit in Stunden (+ Betriebszeit in Jahren = Gesamtbetriebszeit)
 channel-type.valloxmv.uptimehourscurrent.label = Betriebszeit seit letztem Stromausfall (Stunden)
 channel-type.valloxmv.uptimehourscurrent.description = Betriebszeit seit letztem Stromausfall in Stunden
+channel-type.valloxmv.filterchangeddate.label = Letzer Filterwechsel
```
[the remark this answers is not in the document]
Comment (bjoernbrings) [reply addons/binding/org.openhab.binding.valloxmv/ESH-INF/i18n/valloxmv_de.properties]: ups

File: addons/binding/org.openhab.binding.valloxmv/ESH-INF/i18n/valloxmv_de.properties [hunk 5]
```diff
@@ -49,6 +49,10 @@ channel-type.valloxmv.uptimehours.label = Zeit in Betrieb (Stunden)
...
+channel-type.valloxmv.filterchangeddate.description = Datum des letzten Filterwechsels
+channel-type.valloxmv.remainingfilterdays.label = Tage bis Filtertausch
+channel-type.valloxmv.remainingfilterdays.description = Anzahl an Tagen wann der Filter getauscht werden muss
```
[the remark this answers is not in the document]
Comment (bjoernbrings) [reply addons/binding/org.openhab.binding.valloxmv/ESH-INF/i18n/valloxmv_de.properties]: Changed last part of the description to "Anzahl an Tagen bis zum nächsten Filtertausch"

File: addons/binding/org.openhab.binding.valloxmv/ESH-INF/thing/thing-types.xml [hunk 6]
```diff
@@ -172,6 +174,21 @@
 		<state pattern="%d h" readOnly="true" />
 	</channel-type>
 
+	<channel-type id="filterchangeddate" advanced="true">
+		<item-type>DateTime</item-type>
+		<label>Last filter change</label>
```
[the remark this answers is not in the document]
Comment (bjoernbrings) [reply addons/binding/org.openhab.binding.valloxmv/ESH-INF/thing/thing-types.xml]: I did change it, but I didn't use camel case anywhere before ...

File: addons/binding/org.openhab.binding.valloxmv/ESH-INF/thing/thing-types.xml [hunk 7]
```diff
@@ -172,6 +174,21 @@
...
+		<description>Date filter was changed last time</description>
+		<category>Date</category>
+		<state readOnly="true" />
+	</channel-type>
+
+	<channel-type id="remainingfilterdays" advanced="true">
+		<item-type>Number</item-type>
+		<label>Days until filter change</label>
```
[the remark this answers is not in the document]
Comment (bjoernbrings) [reply addons/binding/org.openhab.binding.valloxmv/ESH-INF/thing/thing-types.xml]: Sounds good

File: addons/binding/org.openhab.binding.valloxmv/src/main/java/org/openhab/binding/valloxmv/internal/ValloxMVWebSocket.java (hunk 1 above)
Comment (bjoernbrings) [reply addons/binding/org.openhab.binding.valloxmv/src/main/java/org/openhab/binding/valloxmv/internal/ValloxMVWebSocket.java]: Sounds good

File: addons/binding/org.openhab.binding.valloxmv/ESH-INF/thing/thing-types.xml (hunk 2 above)
Comment (bjoernbrings) [reply addons/binding/org.openhab.binding.valloxmv/ESH-INF/thing/thing-types.xml]: changed

File: addons/binding/org.openhab.binding.valloxmv/ESH-INF/thing/thing-types.xml (hunk 3 above)
Comment (bjoernbrings) [reply addons/binding/org.openhab.binding.valloxmv/ESH-INF/thing/thing-types.xml]: Took the categorie out

(honestly just used an example taken from ntp:

```
	<channel-type id="dateTime-channel">
		<item-type>DateTime</item-type>
		<label>Date</label>
		<description>NTP refreshed date &amp; time</description>
		<category>Date</category>
		<state readOnly="true" />
	</channel-type>
```
)

**Comment by bjoernbrings** [comment]
Thanks for the quick review. Have tried to adjust everything accordingly.

**Comment by bjoernbrings** [comment]
Any idea why DCO complains when I signoff with my real name (checked different PRs and you all signoff with your realname without issues) ...

File: addons/binding/org.openhab.binding.valloxmv/src/main/java/org/openhab/binding/valloxmv/internal/ValloxMVWebSocket.java [hunk 8]
```diff
@@ -321,22 +334,26 @@ public void onBinary(InputStream in) {
...
+                updateChannel(ValloxMVBindingConstants.CHANNEL_CELLSTATE, new DecimalType(iCellstate));
+                updateChannel(ValloxMVBindingConstants.CHANNEL_UPTIME_YEARS, new DecimalType(iUptimeYears));
+                updateChannel(ValloxMVBindingConstants.CHANNEL_UPTIME_HOURS, new DecimalType(iUptimeHours));
                 updateChannel(ValloxMVBindingConstants.CHANNEL_UPTIME_HOURS_CURRENT,
-                        new DecimalType(bdUptimeHoursCurrent));
+                        new DecimalType(iUptimeHoursCurrent));
+                updateChannel(ValloxMVBindingConstants.CHANNEL_FILTER_CHANGED_DATE,
+                        new DateTimeType(ZonedDateTime.ofInstant(cFilterChangedDate.toInstant(), TimeZone.getDefault().toZoneId())));
+                updateChannel(ValloxMVBindingConstants.CHANNEL_REMAINING_FILTER_DAYS,
+                        new DecimalType(iRemainingTimeForFilter));
```
[the remark this answers is not in the document]
Comment (bjoernbrings) [reply addons/binding/org.openhab.binding.valloxmv/src/main/java/org/openhab/binding/valloxmv/internal/ValloxMVWebSocket.java]: Good point

**Comment by bjoernbrings** [comment]
I remember that I did change my username, but that is quite some years ago.

As I do not use git on command line (but egit/git gui). In git gui 'bjoernbrings' is shown. I did change this to 'Björn Brings' hope that will work.

**Pull request merged**